modal/dict.py

```diff
@@ -71,8 +73,26 @@ async def _load(self: _Dict, resolver: Resolver, existing_object_id: Optional[st
     def __init__(self, data={}):
         """mdmd:hidden"""
         deprecation_error((2023, 6, 27), "`Dict({...})` is deprecated. Please use `Dict.new({...})` instead.")
-        obj = _Dict.new(data)
-        self._init_from_other(obj)
+
+    @classmethod
+    @asynccontextmanager
+    async def ephemeral(
```

Comment:
I know we just got _rid_ of the intermediate `StatefulObject` class but the amount of copy-paste we'll need to do to support all of these is still a little unfortunate. Not sure if it's necessarily better but maybe we could have like an `EphemeralMixin` to handle that?

Reply:
I prefer code duplication in this case since in theory these constructor might have different constructor arguments. They also use different protobuf classes and grpc endpoints etc, so it's not trivial to use a mixin.

There's separately a bit too much code duplication elsewhere on the objects though, but imo the `ephemeral` method seems fine to me.